CONTRIBUTING.md

```diff
@@ -25,6 +25,52 @@ The following information is most helpful:
 * **Pointers** — If you have a hunch or just don't want to bother with a pull request,
 please point me in the right direction (e.g. line numbers) to fix it.
 
+## Development, Builds and Testing
+
+### Project Layout
+
+* **package.json** — Metadata about the project, including details for the extensions browser, dependencies, build scripts, commands and user configuration
+* **README.md** — User-facing documetnation for the extension
+* **CHANGELOG.md** — Version history and changes
+* **CONTRIBUTING.md** — This file
+* **LICENSE** — License information about the project
+* **NOTICE.txt** — License information about third party packages
+* **client** — Main extension folder
+    * **extension.ts** — Code insertion point.  Contains initialization code and defines commands callable by user.
+    * **marklogicClient.ts** — Interface to MarkLogic [Node Client API](https://github.com/marklogic/node-client-api) package, which is used to communicate with ML Server.
+    * **marklogic-types.d.ts** — TypeScript type declarations used by the package.
+* **server** — Language server extension folder (provides syntax highlighting; based on vscode-languageserver extension)
+    * **package.json** — As the language server is technically a separate package, this defines the dependencies for it
+    * **server.ts** — Code insertion point.  Contains initialization code for language server.
+
+### Building and Testing Project
+
+Building the project requires [Node.js](https://nodejs.org/) to be installed on your local machine
```

Comment:
Does it matter which version, or just latest LTS version?

Reply:
Since it just caught me up - the version of @types/node in master is 14.14.5.  I just updated that to 14.17.0 in a different branch I'm working on so I could use crypto.randomUUID().  Latest version of 14 is 14.21.1, and that's what I'm using.